Sources/StreamChatUI/ChatMessageList/Attachments/Unsupported/UnsupportedAttachmentViewInjector.swift

```diff
@@ -0,0 +1,54 @@
+//
+// Copyright © 2023 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+/// StreamChat's native attachment types.
+internal extension AttachmentType {
+    static var knownTypes: Set<AttachmentType> {
+        [.image, .file, .giphy, .video, .audio, .voiceRecording, .linkPreview]
+    }
+}
+
+/// Returns all the non-native attachments.
+internal extension ChatMessage {
+    var unsupportedAttachments: [AnyChatMessageAttachment] {
+        allAttachments.filter {
+            AttachmentType.knownTypes.contains($0.type) == false
+        }
+    }
+}
+
+/// The injector for unknown/unsupported attachments.
+///
+/// By default it renders unsupported attachments as file attachments.
+public class UnsupportedAttachmentViewInjector: AttachmentViewInjector {
```

Comment:
Can this have an impact on custom attachments from customers? Since we're introducing a new injector, not everyone will start using that.

Reply:
No impact whatsover 👍